setup/steps/establish_superuser_connection.go

```diff
@@ -0,0 +1,152 @@
+package steps
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strconv"
+
+	survey "github.com/AlecAivazis/survey/v2"
+	"github.com/pganalyze/collector/setup/query"
+	s "github.com/pganalyze/collector/setup/state"
+)
+
+var EstablishSuperuserConnection = &s.Step{
+	Description: "Ensure Postgres superuser connection",
+	Check: func(state *s.SetupState) (bool, error) {
+		if state.QueryRunner == nil {
+			return false, nil
+		}
+		err := state.QueryRunner.PingSuper()
+		return err == nil, err
+	},
+	Run: func(state *s.SetupState) error {
+		localPgs, err := discoverLocalPostgres()
+		if err != nil {
+			return err
+		}
+		var selectedPg LocalPostgres
+		if state.Inputs.Scripted {
+			if !state.Inputs.PGSetupConnPort.Valid {
+				return errors.New("no port specified for setup Postgres connection")
+			}
+			for _, pg := range localPgs {
+				if int(state.Inputs.PGSetupConnPort.Int64) == pg.Port &&
+					(!state.Inputs.PGSetupConnSocketDir.Valid ||
+						state.Inputs.PGSetupConnSocketDir.String == pg.SocketDir) {
+					selectedPg = pg
+					break
+				}
+			}
+			if selectedPg.Port == 0 {
+				var socketDirStr string
+				if state.Inputs.PGSetupConnSocketDir.Valid {
+					socketDirStr = " in " + state.Inputs.PGSetupConnSocketDir.String
+				}
+
+				return fmt.Errorf(
+					"no Postgres server found listening on %d%s",
+					state.Inputs.PGSetupConnPort.Int64,
+					socketDirStr,
+				)
+			}
+		} else {
+			if len(localPgs) == 0 {
+				return errors.New("failed to find a running local Postgres install")
+			} else if len(localPgs) == 1 {
+				selectedPg = localPgs[0]
```

Comment:
Might be good if we move this up, so its also run for scripted installs (i.e. clear error message when there are no local Postgres installs, and don't require specifying a port when there is only a single Postgres install detected)